data-azure-cosmos/src/main/java/io/micronaut/data/cosmos/operations/DefaultReactiveCosmosRepositoryOperations.java

```diff
@@ -0,0 +1,712 @@
+                return result.mapNotNull(item -> {
+                    if (resultType.isInstance(item)) {
+                        return (R) item;
+                    } else if (item != null) {
+                        return ConversionService.SHARED.convertRequired(item, resultType);
+                    }
+                    return null;
+                });
+            }
+        } catch (Exception e) {
+            return Flux.error(new DataAccessException("Cosmos SQL Error executing Query: " + e.getMessage(), e));
+        }
+    }
+
+    @Override
+    @NonNull
+    public <T> Mono<T> persist(@NonNull InsertOperation<T> operation) {
+        CosmosAsyncContainer container = getContainer(operation);
+        Class<T> rootEntity = operation.getRootEntity();
+        RuntimePersistentEntity<T> persistentEntity = runtimeEntityRegistry.getEntity(rootEntity);
+        CosmosReactiveOperationContext<T> ctx = new CosmosReactiveOperationContext<>(operation.getAnnotationMetadata(),
+            operation.getRepositoryType(), container, rootEntity);
+        CosmosReactiveEntityOperation<T> op = createCosmosInsertOneOperation(ctx, persistentEntity, operation.getEntity());
+        op.persist();
+        return op.getEntity();
+    }
+
+    @Override
+    @NonNull
+    public <T> Mono<T> update(@NonNull UpdateOperation<T> operation) {
+        CosmosAsyncContainer container = getContainer(operation);
+        Class<T> rootEntity = operation.getRootEntity();
+        RuntimePersistentEntity<T> persistentEntity = runtimeEntityRegistry.getEntity(rootEntity);
+        CosmosReactiveOperationContext<T> ctx = new CosmosReactiveOperationContext<>(operation.getAnnotationMetadata(),
+            operation.getRepositoryType(), container, rootEntity);
+        CosmosReactiveEntityOperation<T> op = createCosmosReactiveReplaceItemOperation(ctx, persistentEntity, operation.getEntity());
+        op.update();
+        return op.getEntity();
+    }
+
+    @Override
+    @NonNull
+    public  <T> Flux<T> updateAll(UpdateBatchOperation<T> operation) {
+        Class<T> rootEntity = operation.getRootEntity();
+        RuntimePersistentEntity<T> persistentEntity = runtimeEntityRegistry.getEntity(rootEntity);
+        CosmosAsyncContainer container = getContainer(persistentEntity);
+        CosmosReactiveOperationContext<T> ctx = new CosmosReactiveOperationContext<>(operation.getAnnotationMetadata(),
+            operation.getRepositoryType(), container, rootEntity);
+        CosmosReactiveEntitiesOperation<T> op = createCosmosReactiveBulkOperation(ctx, persistentEntity, operation, BulkOperationType.UPDATE);
+        op.update();
+        return op.getEntities();
+    }
+
+    @Override
+    @NonNull
+    public <T> Flux<T> persistAll(InsertBatchOperation<T> operation) {
+        Flux<T> result = Flux.empty();
+        for (InsertOperation<T> insertOperation : operation.split()) {
```

Comment:
This should be batched in the future.

Reply:
Done now in https://github.com/micronaut-projects/micronaut-data/pull/1797/commits/45b055614910734841988465021e7fe2fd9c800a